File: m-solutions2020/m-solutions2020_d.cs
// https://atcoder.jp/contests/m-solutions2020/submissions/15466723

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var N = int.Parse(Console.ReadLine());
        var A = Console.ReadLine().Split().Select(long.Parse).ToArray();

        var money = 1000L;
        for (int i = 0; i < N - 1; i++)
        {
            money += A[i] < A[i + 1] ? (A[i + 1] - A[i]) * (money / A[i]) : 0;
        }
        Console.WriteLine(money);
    }
}

File: m-solutions2020/m-solutions2020_d.fs
// https://atcoder.jp/contests/m-solutions2020/submissions/15465914

let N = stdin.ReadLine() |> int
let A = stdin.ReadLine().Split() |> Array.map int64

let rec solver money i =
    if i = N - 1 then money
    elif A.[i] < A.[i + 1] then
        let stocks = money / A.[i]
        solver (money + (A.[i + 1] - A.[i]) * stocks) (i + 1)
    else solver money (i + 1)

solver 1000L 0 |> stdout.WriteLine
